From: Philip Kuryloski <kuryloskip@vmware.com>
Subject: Fix `rabbitmqctl status` when the disk_free cannot be
 determined

By adjusting RabbitMQ.CLI.InformationUnit to no longer attempt to
convert non-numeric values

Origin: upstream, https://github.com/rabbitmq/rabbitmq-server/commit/9dc9b92a94f3f3bb48824ea21b9f4a9236a9a3d3
Bug-Ubuntu: https://bugs.launchpad.net/bugs/1999816
Last-Update: 2023-01-30
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
diff --git a/deps/rabbitmq_cli/lib/rabbitmq/cli/information_unit.ex b/deps/rabbitmq_cli/lib/rabbitmq/cli/information_unit.ex
index c5e02277a9e..fa548c8fe18 100644
--- a/deps/rabbitmq_cli/lib/rabbitmq/cli/information_unit.ex
+++ b/deps/rabbitmq_cli/lib/rabbitmq/cli/information_unit.ex
@@ -30,14 +30,18 @@ defmodule RabbitMQ.CLI.InformationUnit do
     :rabbit_resource_monitor_misc.parse_information_unit(val)
   end
 
-  def convert(bytes, "bytes") do
+  def convert(bytes, "bytes") when is_number(bytes) do
     bytes
   end
 
-  def convert(bytes, unit) do
+  def convert(bytes, unit) when is_number(bytes) do
     do_convert(bytes, String.downcase(unit))
   end
 
+  def convert(:unknown, _) do
+    :unknown
+  end
+
   def known_unit?(val) do
     MapSet.member?(known_units(), String.downcase(val))
   end
